go/vt/vtgate/planbuilder/delete.go

```diff
@@ -144,16 +144,7 @@ func checkIfDeleteSupported(del *sqlparser.Delete, semTable *semantics.SemTable)
 		return semTable.NotUnshardedErr
 	}
 
-	// Delete is only supported for a single TableExpr which is supposed to be an aliased expression
-	multiShardErr := vterrors.VT12001("multi-shard or vindex write statement")
-	if len(del.TableExprs) != 1 {
-		return multiShardErr
-	}
-	_, isAliasedExpr := del.TableExprs[0].(*sqlparser.AliasedTableExpr)
-	if !isAliasedExpr {
-		return multiShardErr
-	}
-
+	// Delete is only supported for single Target.
```

Comment:
Can there be no target at all?

Reply:
there there is no nothing to delete. So, one target is a must.